enable_overlayfs.sh

```
@@ -0,0 +1,45 @@
+# Enable Raspberry Pi OverlayFS root in a mounted image.
+#
+# raspi-config's do_overlayfs uses `update-initramfs -c -k "$(uname -r)"`,
+# but under qemu-user chroot `uname -r` returns the host kernel, so the
+# command fails silently and the image ships without /boot/initrd.img.
+# This script regenerates the initramfs against the image's actual kernel
+# and ensures cmdline.txt / config.txt are set up for overlay boot.
+set -euxo pipefail
+
+mount=$1
+
+kver=$(ls "${mount}/lib/modules" 2>/dev/null | sort -V | tail -n1 || true)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)